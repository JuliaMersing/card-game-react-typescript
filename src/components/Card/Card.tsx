import React from 'react';
import { CardData } from '../services/api';

export const Card: React.FunctionComponent<CardData> = ({
	code,
	image,
}: CardData) => {
	return <img src={image} alt={code} className="cards-image" />;
};
