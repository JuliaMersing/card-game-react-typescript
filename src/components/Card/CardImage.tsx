import React from 'react';

interface CardImageProps {
	src: string;
	alt: string;
}

export const CardImage: React.FunctionComponent<CardImageProps> = ({
	src,
	alt,
}: CardImageProps) => {
	return <img src={src} alt={alt} className="max-w-xs" />;
};
