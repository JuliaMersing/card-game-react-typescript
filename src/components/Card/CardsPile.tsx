import React from 'react';
import { Card } from './Card';
import { CardData } from '../services/api';

interface CardsPileProps {
	cards: CardData[];
}

export const CardsPile: React.FC<CardsPileProps> = ({
	cards,
}: CardsPileProps) => {
	const cardsList = cards.map((card) => {
		return (
			<li key={card.code}>
				<Card image={card.image} code={card.code} />
			</li>
		);
	});

	return <ul className="cards-list">{cardsList}</ul>;
};
