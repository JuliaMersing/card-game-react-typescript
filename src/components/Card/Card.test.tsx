import { render, screen } from '@testing-library/react';
import { Card } from './Card';

describe('Card component', () => {
	test('should match snapshot', () => {
		render(
			<Card
				code={'6H'}
				image={'https://deckofcardsapi.com/static/img/6H.png'}
			/>
		);
		const card = screen.getByAltText(/6H/i);
		expect(card).toMatchSnapshot();
	});
});
