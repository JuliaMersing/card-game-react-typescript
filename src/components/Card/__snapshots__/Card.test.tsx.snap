// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `
<img
  alt="6H"
  class="cards-image"
  src="https://deckofcardsapi.com/static/img/6H.png"
/>
`;
